.github/workflows/build.yml

```diff
@@ -27,13 +27,42 @@ jobs:
             - run: yarn lint
     unit:
         runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node: [ 14, 16 ]
         steps:
             - uses: actions/setup-node@v1
               with:
-                  node-version: 14
+                  node-version: ${{ matrix.node }}
 
             - uses: actions/checkout@v2
 
             - run: yarn
 
             - run: yarn test:unit
+
+    e2e:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                node: [ 14, 16 ]
+                testCmd: [ 'e2e_ganache', 'e2e_geth' ]
+        env:
+            TEST_CMD: ${{ matrix.testCmd }}
+        steps:
+            -   uses: actions/setup-node@v1
+                with:
+                    node-version: ${{ matrix.node }}
+
+            -   uses: actions/checkout@v2
+
+            -   run: yarn
+
+            -   if: (env.TEST_CMD == 'e2e_ganache')
+                run: yarn ganache-cli-background
```

Comment:
Probably best to use something like [ganche.sh](https://github.com/ChainSafe/web3.js/blob/1.x/scripts/e2e.ganache.sh) as it handles cleanup (needs to be updated to use [ganche](https://github.com/trufflesuite/ganache) as `ganche-cli` was deprecated)